library/core/class.controller.php

```diff
@@ -2142,4 +2164,23 @@ public function title($Title = null, $Subtitle = null) {
 
         return $this->data('Title');
     }
+
+    /**
+     * Set the destination URL where the page will be redirected after an ajax request.
+     *
+     * @param string|null $destination Destination URL or path.
+     *      Redirect to current URL if nothing or null is supplied.
+     * @param bool $trustedOnly Non trusted destinations will be redirected to /home/leaving?Target=$destination
+     */
+    public function redirectTo($destination = null, $trustedOnly = true) {
+        if ($destination === null) {
+            $url = url('');
+        } elseif ($trustedOnly) {
+            $url = safeURL($destination);
+        } else {
+            $url = url($destination);
+        }
+
+        $this->redirectTo = $url;
```

Comment:
For this to truly replace the existing direct assignment addons are using for setting the `RedirectUrl` property, shouldn't this function also be setting it? This seems like we're breaking backwards compatibility. If some addon out there is relying on `RedirectUrl` having been set at some point where this function is being introduced, it would begin to fail. I know we're trying to deprecate usage of the `RedirectUrl` property, but there has to be a better way than outright removing a public property being set in so many places.